libc/string/string_tokpbrkstr.c

```diff
@@ -0,0 +1,1029 @@
+
+TEST_GROUP(string_tok);
+TEST_GROUP(string_tok_r);
+TEST_GROUP(string_str);
+TEST_GROUP(string_pbrk);
+
+static char *create_extAscii_set(void)
+{
+	int i;
+	char *hold = malloc(EXTENDED_ASCII_LENGTH);
+
+	for (i = 1; i < EXTENDED_ASCII_LENGTH; i++) {
+		hold[i - 1] = i;
+	}
+	hold[i - 1] = 0;
+
+	return hold;
+}
+
+
+TEST_SETUP(string_tok)
+{
+}
+
+
+TEST_TEAR_DOWN(string_tok)
+{
+}
+
+
+TEST(string_tok, basic)
+{
+	char str[] = "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas eleifend elementum tellum.";
+	const char *separators = ", .";
+	char *token;
+
+	TEST_ASSERT_EQUAL_PTR(str, (token = strtok(str, separators)));
+	TEST_ASSERT_EQUAL_STRING("Lorem", token);
+
+	TEST_ASSERT_EQUAL_PTR(&str[6], (token = strtok(NULL, separators)));
+	TEST_ASSERT_EQUAL_STRING("ipsum", token);
+
+	TEST_ASSERT_EQUAL_PTR(&str[12], (token = strtok(NULL, separators)));
+	TEST_ASSERT_EQUAL_STRING("dolor", token);
+
+	TEST_ASSERT_EQUAL_PTR(&str[18], (token = strtok(NULL, separators)));
+	TEST_ASSERT_EQUAL_STRING("sit", token);
+
+	TEST_ASSERT_EQUAL_PTR(&str[22], (token = strtok(NULL, separators)));
+	TEST_ASSERT_EQUAL_STRING("amet", token);
+
+	TEST_ASSERT_EQUAL_PTR(&str[28], (token = strtok(NULL, separators)));
+	TEST_ASSERT_EQUAL_STRING("consectetur", token);
+
+	TEST_ASSERT_EQUAL_PTR(&str[40], (token = strtok(NULL, separators)));
+	TEST_ASSERT_EQUAL_STRING("adipiscing", token);
+
+	TEST_ASSERT_EQUAL_PTR(&str[51], (token = strtok(NULL, separators)));
+	TEST_ASSERT_EQUAL_STRING("elit", token);
+
+	TEST_ASSERT_EQUAL_PTR(&str[57], (token = strtok(NULL, separators)));
+	TEST_ASSERT_EQUAL_STRING("Maecenas", token);
+
+	TEST_ASSERT_EQUAL_PTR(&str[66], (token = strtok(NULL, separators)));
+	TEST_ASSERT_EQUAL_STRING("eleifend", token);
+
+	TEST_ASSERT_EQUAL_PTR(&str[75], (token = strtok(NULL, separators)));
+	TEST_ASSERT_EQUAL_STRING("elementum", token);
+
+	TEST_ASSERT_EQUAL_PTR(&str[85], (token = strtok(NULL, separators)));
+	TEST_ASSERT_EQUAL_STRING("tellum", token);
+
+	TEST_ASSERT_EQUAL_PTR(NULL, strtok(NULL, separators));
+}
+
+
+/* assert that looking for tokens stops after encountering even one NUL term */
+TEST(string_tok, torn)
+{
+	char str[] = "Lor\0em ipsum";
+	const char *separators = " ";
+	char *tok;
+	TEST_ASSERT_EQUAL_PTR(str, (tok = strtok(str, separators)));
+	TEST_ASSERT_EQUAL_STRING("Lor", tok);
+	TEST_ASSERT_EQUAL_PTR(NULL, (tok = strtok(NULL, separators)));
+}
+
+
+TEST(string_tok, empty_args)
+{
+	/* Different scenarios of acquiring null pointer as output*/
+	TEST_ASSERT_EQUAL_PTR(NULL, strtok("", "d"));
+	TEST_ASSERT_EQUAL_PTR(NULL, strtok(NULL, ""));
+
+	TEST_ASSERT_EQUAL_PTR(NULL, strtok("\0", ""));
+	TEST_ASSERT_EQUAL_PTR(NULL, strtok(NULL, ""));
+
+	TEST_ASSERT_EQUAL_STRING("abc", strtok("abc", ""));
+}
+
+
+TEST(string_tok, multi_call)
+{
+	char loremStr[20] = LOREM_IPSUM,
+		 sep[5] = { 0 },
+		 multiCallStr[ASCII_LENGTH] = { 0 },
+		 *asciiStr = testdata_createCharStr(ASCII_LENGTH);
```

Comment:
Please switch to the following approach in this test and prepare separate PR with the changes in the rest of files.

```suggestion
	char loremStr[20] = LOREM_IPSUM;
	char sep[5] = { 0 };
	char multiCallStr[ASCII_LENGTH] = { 0 };
	char *asciiStr = testdata_createCharStr(ASCII_LENGTH);
```